pkg/operator/status/status_controller.go

```diff
@@ -144,10 +147,10 @@ func (c StatusSyncer) sync() error {
 	}
 
 	clusterOperatorObj.Status.RelatedObjects = c.relatedObjects
-	configv1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, unionInertialCondition("Degraded", operatorv1.ConditionFalse, currentDetailedStatus.Conditions...))
-	configv1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, unionCondition("Progressing", operatorv1.ConditionFalse, currentDetailedStatus.Conditions...))
-	configv1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, unionCondition("Available", operatorv1.ConditionTrue, currentDetailedStatus.Conditions...))
-	configv1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, unionCondition("Upgradeable", operatorv1.ConditionTrue, currentDetailedStatus.Conditions...))
+	configv1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, unionCondition("Degraded", operatorv1.ConditionFalse, c.inertia, currentDetailedStatus.Conditions...))
+	configv1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, unionCondition("Progressing", operatorv1.ConditionFalse, c.inertia, currentDetailedStatus.Conditions...))
```

Comment:
I prefer no intertia on these for now.  just pass nil please.

Reply:
> I prefer no intertia on these for now. just pass nil please.

Does library-go need to have an opinion here?  Seems like folks could set up their `Inertia` implementation to [restrict themselves to Degraded][1] if they wanted.

[1]: https://github.com/openshift/library-go/pull/632#issuecomment-563359300

Reply:
> Does library-go need to have an opinion here?

Talked this over out of band.  Issues considered:

* There's currently no use case for inertia in the other unioned cases.  Availability, progressing, and upgradeable should all switch over immediately.
* If someone thinks up a case, they'll need to argue it out in a library-go PR, where library-go maintainers can decide if the case makes sense or if the user is misunderstanding the union/type semantics.
* If the case is convincing, and library-go adds a new `WithAvailableInertia` (or whatever), existing `WithDegradedInertia` can bump to the new library-go without breaking.

So I've pushed 60ab2f004 -> 68a4d3905 leaving inertia out of `NewClusterOperatorStatusController`, adding a new, future-proof `WithDegradedInertia`, and setting `inertia` `nil` for these other types.